contracts/dao/voting/vesting-lp-vault/src/contract.rs

```diff
@@ -0,0 +1,331 @@
+        .add_attribute("atom_oracle_contract", config.atom_oracle_contract)
+        .add_attribute("usdc_vesting_lp_contract", config.usdc_vesting_lp_contract)
+        .add_attribute("usdc_oracle_contract", config.usdc_oracle_contract)
+        .add_attribute(
+            "manager",
+            config
+                .manager
+                .map(|a| a.to_string())
+                .unwrap_or_else(|| "None".to_string()),
+        ))
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn execute(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: ExecuteMsg,
+) -> ContractResult<Response> {
+    match msg {
+        ExecuteMsg::Bond {} => execute_bond(deps, env, info),
+        ExecuteMsg::Unbond { amount } => execute_unbond(deps, env, info, amount),
+        ExecuteMsg::UpdateConfig {
+            owner,
+            atom_vesting_lp_contract,
+            atom_oracle_contract,
+            usdc_vesting_lp_contract,
+            usdc_oracle_contract,
+            manager,
+            name,
+            description,
+        } => execute_update_config(
+            deps,
+            info,
+            owner,
+            atom_vesting_lp_contract,
+            atom_oracle_contract,
+            usdc_vesting_lp_contract,
+            usdc_oracle_contract,
+            manager,
+            name,
+            description,
+        ),
+    }
+}
+
+pub fn execute_bond(_deps: DepsMut, _env: Env, _info: MessageInfo) -> ContractResult<Response> {
+    Err(ContractError::BondingDisabled {})
+}
+
+pub fn execute_unbond(
+    _deps: DepsMut,
+    _env: Env,
+    _info: MessageInfo,
+    _amount: Uint128,
+) -> ContractResult<Response> {
+    Err(ContractError::DirectUnbondingDisabled {})
+}
+
+#[allow(clippy::too_many_arguments)]
+pub fn execute_update_config(
+    deps: DepsMut,
+    info: MessageInfo,
+    new_owner: String,
+    new_atom_vesting_lp_contract: String,
+    new_atom_oracle_contract: String,
+    new_usdc_vesting_lp_contract: String,
+    new_usdc_oracle_contract: String,
+    new_manager: Option<String>,
+    new_name: String,
+    new_description: String,
+) -> ContractResult<Response> {
+    let mut config: Config = CONFIG.load(deps.storage)?;
+    if info.sender != config.owner && Some(info.sender.clone()) != config.manager {
+        return Err(ContractError::Unauthorized {});
+    }
+
+    let new_owner = deps.api.addr_validate(&new_owner)?;
+    let new_atom_vesting_lp_contract = deps.api.addr_validate(&new_atom_vesting_lp_contract)?;
+    let new_atom_oracle_contract = deps.api.addr_validate(&new_atom_oracle_contract)?;
+    let new_usdc_vesting_lp_contract = deps.api.addr_validate(&new_usdc_vesting_lp_contract)?;
+    let new_usdc_oracle_contract = deps.api.addr_validate(&new_usdc_oracle_contract)?;
+    let new_manager = new_manager
+        .map(|new_manager| deps.api.addr_validate(&new_manager))
+        .transpose()?;
+
+    if info.sender != config.owner && new_owner != config.owner {
```

Comment:
Can't wait for most of this code to disappear with the `manager` in the next pr tbh